databuilder/query_engines/in_memory.py

```diff
@@ -134,6 +134,14 @@ def visit_Sum(self, node):
         col = self.visit(node.source)
         return col.aggregate_values(sum, default=None)
 
+    def visit_Mean(self, node):
+        def mean(values):
+            # NULLs are already handled by `aggregate_values`
+            return sum(values) / len(values)
```

Comment:
Why not use [`statistics.mean`](https://docs.python.org/3.10/library/statistics.html#statistics.mean) or [`statistics.fmean`](https://docs.python.org/3.10/library/statistics.html#statistics.fmean) instead?

Reply:
Oh, you're quite right! When I first wrote this I thought I had to do all the NULL handling myself so I wrote my own `mean()` function. It was only when I got the coverage failure that I realised that `aggregate_values()` already does the NULL handling, so I ripped that out but didn't think to replace the function entirely.

`statistics.fmean` does what we want here because it gives a float result even if all the inputs are ints and happen to by divisible by the count.